pkg/systemd/services/services.jsx

```diff
@@ -258,6 +258,12 @@ class ServicesPage extends React.Component {
            (this.seenPaths.size == 0 || this.seenPaths.size > Object.keys(nextState.unit_by_path).length))
             return false;
 
+        /* XXX - do a complete shallow compare.  This here already
+         * makes testBasic pass.
+         */
+        if (this.state.currentTextFilter != nextState.currentTextFilter)
+            return true;
```

Comment:
Does that still cause the actual filtering and table update to get throttled? That's the #1 case of when to use throttle-debounce, to not synchronously act to keyboard input.

Reply:
We need to act synchronously on keyboard input since we use a controlled component.  If we don't we might miss input events.

Reply:
We would need to decouple the state for the actual list from the rest of the state.  That would be a good improvement all around, I'd say.